app/mailers/survey_mailer.rb

```diff
@@ -0,0 +1,61 @@
+##
+# Mailer to deliver a survey to users who have recently made a request for
+# information
+#
+class SurveyMailer < ApplicationMailer
+  include AlaveteliFeatures::Helpers
+
+  before_action :set_footer_template
+
+  def survey_alert(info_request)
+    return unless Survey.enabled?
+
+    @info_request = info_request
+    @url = Survey.new(info_request.public_body).url
```

Comment:
Now I've seen your implementation, I like that we encapsulate rather than pass the info request as an argument to `url`.

I think it would be clearer to do `@url = Survey.new(info_request).url` though. We're creating a survey _about the request_ – not about the _authority_. This wouldn't change the internal implementation much, and would make it easier to add other parameters (say, `?embargoed=1`) if we wanted to.

I can live with it as it is though.

Reply:
Lets leave it as it is for now. Felt wrong to pass into `Survey.new` the `info_request` and then only be calling `.public_body` on it. Simpel change if we ever need to change the survey URL in the future